### PR TITLE
Add docs for Monitoring and Alerting of routers

### PR DESCRIPTION
**Pull request by krithika369**
Adding User / Developer docs related to monitoring and alerting.

### CODE REVIEW

File: docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+* [Knative](https://knative.dev/docs/serving/observability/metrics/serving-metrics/#autoscaler)
+* [Istio](https://istio.io/v1.9/docs/tasks/observability/metrics/)
+
+The following custom Prometheus metrics are published by the Turing routers too.
+
+| Metric Name | Description | Type | Tags | Unit |
+| ----------- | ----------- | ---- | ---- | ---- |
+| mlp_turing_exp_engine_request_duration_ms | The duration for fetching a treatment from the experiment engine | Histogram | `status`, `engine` | Milliseconds |
+| mlp_route_request_duration_ms | The duration for the call to a route | Histogram | `status`, `route` | Milliseconds |
+| mlp_turing_comp_request_duration_ms | The duration for a custom operation in the code, useful for debugging | Histogram | `status`, `component` | | Milliseconds |
```
Comment (deadlycoconuts) [inline docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md:15]: I might be saying something silly but isn't it meant to be '*complete* operation' and not 'custom operation'?

**Review by deadlycoconuts — APPROVED** [review]
Looks good to be overall, I just have one tiny comment otherwise it should be good to go 🚀 Thanks a lot for the detailed write up!

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -19,12 +19,21 @@
...
+    * [Redeploy version from version details page](how-to/redeploy-a-router/redeploy-from-version-detail.md)
+* [Deleting routers](how-to/delete-a-router/README.md)
+    * [Deleting router versions](how-to/delete-a-router/delete-a-router-version.md)
+    * [Deleting router versions from details page](how-to/delete-a-router/delete-a-router-version-from-details-page.md)
+    * [Deleting routers](how-to/delete-a-router/delete-a-router.md)
+
+## Developer Guides
+* [Getting started](dev-docs/README.md)
+* [Monitoring and alerting](dev-docs/monitoring-and-alerting/README.md)
+    * [Configure a monitoring backend](dev-docs/monitoring-and-alerting/configure-monitoring-backend.md.md)
```
Comment (terryyylim) [inline docs/SUMMARY.md:38]: Extra `.md` for file extension. Likewise for the row directly below.

File: docs/dev-docs/monitoring-and-alerting/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Monitoring and alerting
+
+Turing exposes a convenient user interface and provides a framework to enable fine-grained monitoring and alerting on the deployed resources (routers, ensemblers, etc.). The  sub-sections provide details on how these features can be made available to the Turing application.
```
Comment (terryyylim) [inline docs/dev-docs/monitoring-and-alerting/README.md:3]: nit: Extra space before `sub-sections`.

File: docs/dev-docs/monitoring-and-alerting/configure-alerting-backend.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            service_name: turing-hello-turing-router
+            severity: warning
+          annotations:
+            dashboard: http://monitoring.com/turing-dashboard?var-cluster=test-kube-cluster&var-project=test-project&var-experiment=turing-hello
+            description: 'cpu_util for the past 5m: {{ $value }}%'
+            playbook: http://docs.com/Alert+Troubleshooting+Playbook
+            summary: 'cpu_util is higher than the threshold: 90%'
+        - alert: turing-hello-turing-router_cpu_util_violation_development
+          expr: |-
+            sum by(cluster) (rate(container_cpu_usage_seconds_total{environment="staging",pod=~"turing-hello-turing-router-[0-9]*.*"}[1m])) / sum by(cluster) (kube_pod_container_resource_requests_cpu_cores{environment="staging",pod=~"turing-hello-turing-router-[0-9]*.*"}) * 100 > 95
```
Comment (terryyylim) [inline docs/dev-docs/monitoring-and-alerting/configure-alerting-backend.md:40]: Based on this recent [PR](https://github.com/gojek/turing/pull/209) which updated the deprecated metrics, I think we should update this query as well?

File: docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+Users are also free to publish their own custom metrics from the Enricher / Ensembler. All custom metrics (from the router, enricher or ensembler) should be scraped from the `user-container` pods for use.
+
+### Configuring the Monitoring URL on Turing
+
+Once the required dashboard has been created using the Prometheus metrics and other data, the template string `RouterDefaults.MonitoringURLFormat` can be used to configure the monitoring URL, when deploying the Turing application (please refer to the [sample Helm values file](https://github.com/gojek/turing/blob/main/api/turing/config/example.yaml) for an example). This URL will be used by Turing when creating / editing a router and subsequently, in the Turing UI as a navigation link to the dashboard.
+
+## Monitoring Batch Ensemblers
+
+TBD
```
Comment (terryyylim) [inline docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md:25]: Would this section be added in a future PR? Should we prefix a TODO?

File: docs/how-to/monitor-a-router/README.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Monitoring a Router
+
+Turing provides a convenient user interface for monitoring the router deployments and getting the standard metrics out of the box, and configuring alerts based on the performance of the routers. To take advantage of these features, an appropriate monitoring / alerting backend should have been configured.
```
Comment (terryyylim) [inline docs/how-to/monitor-a-router/README.md:3]: super nit: Should we standardize `out of the box` as `out-of-the-box` everywhere in the docs? I noticed `out-of-the-box` being used in `docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md`.

File: docs/how-to/monitor-a-router/monitor-router-performance.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+This document assumes that an appropriate monitoring solution has been integrated to the deployment of Turing.
+The actual functionality may be different, according to the deployment configurations.
+{% endhint %}
+
+1. Navigate to the Router Details View of your router.
+
+2. Click on the monitoring link of your router under Router Actions.
+![monitoring_tab](../../.gitbook/assets/monitoring_tab.png)
+
+3. This link will bring you to the Lens dashboard with standard router metrics such as
```
Comment (terryyylim) [inline docs/how-to/monitor-a-router/monitor-router-performance.md:13]: I think we should not mention `Lens` here. 😅

**Review by terryyylim — CHANGES_REQUESTED** [review]
Thanks for adding the docs @krithika369. Overall LGTM, left some comments to be addressed!

File: docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md (hunk 1 above)
Comment (krithika369) [reply docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md:15]: It is supposed to be "custom" - i.e., some intermediate steps in the request flow (eg: write outcome log).

File: docs/how-to/monitor-a-router/monitor-router-performance.md (hunk 7 above)
Comment (krithika369) [reply docs/how-to/monitor-a-router/monitor-router-performance.md]: Done. Thanks for catching!

File: docs/dev-docs/monitoring-and-alerting/configure-alerting-backend.md (hunk 4 above)
Comment (krithika369) [reply docs/dev-docs/monitoring-and-alerting/configure-alerting-backend.md]: Yes, done.

File: docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md (hunk 5 above)
Comment (krithika369) [reply docs/dev-docs/monitoring-and-alerting/configure-monitoring-backend.md:25]: Changed TBD to TODO. We do need to add this in the future, but it's not urgent now.

**Comment by krithika369** [comment]
Thanks for the reviews! I've addressed the comments now.

File: docs/how-to/monitor-a-router/monitor-router-performance.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# Monitoring Router Performance
+
+{% hint style="warning" %}
+This document assumes that an appropriate monitoring solution has been integrated to the deployment of Turing.
```
Comment (terryyylim) [inline docs/how-to/monitor-a-router/monitor-router-performance.md:4]: super nit: `integrated with the Turing deployment`

**Review by terryyylim — APPROVED** [review]
LGTM! Thanks for the changes @krithika369 ! 🚀

**Pull request merged**